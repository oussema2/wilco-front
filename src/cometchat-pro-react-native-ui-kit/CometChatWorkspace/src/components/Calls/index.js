export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
