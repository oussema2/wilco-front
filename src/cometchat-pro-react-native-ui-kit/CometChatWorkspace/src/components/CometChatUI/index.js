/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
