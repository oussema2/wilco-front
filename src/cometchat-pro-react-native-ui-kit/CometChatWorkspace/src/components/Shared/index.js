export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
