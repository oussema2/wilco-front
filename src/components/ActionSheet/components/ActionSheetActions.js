import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
