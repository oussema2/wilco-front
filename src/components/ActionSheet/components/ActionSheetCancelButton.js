import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
