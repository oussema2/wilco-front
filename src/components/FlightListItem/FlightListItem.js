import React from 'react';
import { View, Text } from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import { CheckBox } from 'react-native-elements';
import FlightToDisplay from '../../entitiesToDisplay/FlightsToDisplay';
import { Image } from '../Image';
import { arrowRight } from '../../assets/icons';
import { styles } from './FlightListItem.styles';
import { palette } from '../../Theme';

const FlightListItem = ( {
	testID, flight, isSelected, onFlightSelected, showCheckBox
} ) => (
	<View testID={testID} style={styles.containerView}>
		<View style={styles.innerContainerView}>
			{showCheckBox && (
				<CheckBox
					containerStyle={styles.checkBox}
					size={24}
					checkedColor={palette.primary.darkCyan}
					checkedIcon="dot-circle-o"
					uncheckedIcon="circle-o"
					checked={isSelected}
					onPress={onFlightSelected}
				/>
			)}
			<View style={styles.originAndDestination}>
				<Text testID="departureAirport-Text" style={styles.airportName} numberOfLines={1}>{flight.from}</Text>
				<View style={styles.arrowIcon}>
					<Image testID="rightArrow-Image" source={arrowRight} size={15} tintColor={palette.grayscale.aluminum} />
				</View>
				<Text testID="arrivalAirport-Text" style={styles.airportName} numberOfLines={1}>{flight.to}</Text>
			</View>
			<View style={styles.timeView}>
				<Text style={styles.time} numberOfLines={1}>{flight.departureTime}</Text>
				<Text testID="hourSeparator-Text" style={styles.timeSeparatorView}>â€¢</Text>
				<Text style={styles.time} numberOfLines={1}>{flight.arrivalTime}</Text>
			</View>
			<Text style={styles.date}>{flight.shortDate}</Text>
		</View>
	</View>
);

FlightListItem.propTypes = {
	testID: PropTypes.string,
	flight: PropTypes.instanceOf( FlightToDisplay ).isRequired,
	isSelected: PropTypes.bool,
	onFlightSelected: PropTypes.func,
	showCheckBox: PropTypes.bool
};

FlightListItem.defaultProps = {
	testID: 'flightListItem-Component',
	isSelected: false,
	onFlightSelected: () => {},
	showCheckBox: true
};

export default observer( FlightListItem );
