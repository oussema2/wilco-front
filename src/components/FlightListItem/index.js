export { default as FlightListItem } from './FlightListItem';
