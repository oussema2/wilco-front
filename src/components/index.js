export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
