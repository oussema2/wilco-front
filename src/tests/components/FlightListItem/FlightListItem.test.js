import React from 'react';
import { render } from '@testing-library/react-native';
import { FlightListItem } from '../../../components/FlightListItem';
import FlightFactory from '../../factories/FlightFactory';
import FlightToDisplay from '../../../entitiesToDisplay/FlightsToDisplay';

describe( '<FlightListItem />', () => {
	let component;
	const testID = 'testing-FlightListItem-Component';
	const flight = FlightFactory.build();
	const flightToDisplay = new FlightToDisplay( { flight } );

	const setUp = () => {
		component = render(
			<FlightListItem testID={testID} flight={flightToDisplay} />
		);
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the FlightListItem component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component.queryByTestId( 'departureAirport-Text' ) ).toHaveTextContent(
				flight.from
			);
			expect( component.queryByTestId( 'rightArrow-Image' ) ).toBeDefined();
			expect( component.queryByTestId( 'arrivalAirport-Text' ) ).toHaveTextContent(
				flight.to
			);
			expect( component.queryByTestId( 'hourSeparator-Text' ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
