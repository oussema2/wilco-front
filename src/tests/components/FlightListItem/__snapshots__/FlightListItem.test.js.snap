// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightListItem /> with the default props renders the FlightListItem component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 40,
      "marginTop": 4,
    }
  }
  testID="testing-FlightListItem-Component"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 0,
          "width": 20,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              Object {
                "minWidth": 24,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "maxWidth": 55,
            "textTransform": "uppercase",
          }
        }
        testID="departureAirport-Text"
      >
        KHPN
      </Text>
      <View
        style={
          Object {
            "marginHorizontal": 4,
            "marginVertical": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 15,
                "width": 15,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../src/assets/icons/arrowRight.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="rightArrow-Image"
            tintColor="#A0A0A0"
          />
        </View>
      </View>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "maxWidth": 55,
            "textTransform": "uppercase",
          }
        }
        testID="arrivalAirport-Text"
      >
        JFK
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#767676",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
          }
        }
      >
        22:41
      </Text>
      <Text
        style={
          Object {
            "color": "#5B5B5B",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 13,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "marginHorizontal": 8,
          }
        }
        testID="hourSeparator-Text"
      >
        •
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#767676",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
          }
        }
      >
        23:42
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#767676",
          "fontFamily": "NotoSansJP-Regular",
          "fontSize": 14,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "textAlign": "right",
        }
      }
    >
      06/29
    </Text>
  </View>
</View>
`;
